src/movies.js

```diff
@@ -1,22 +1,99 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data")
+
+function getAllDirectors() {
+  const directors = movies.map(function (film) {
+    return film.director
+    })
+  return directors
+  }
+  console.log(getAllDirectors(movies).length) // => 250
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectorsUniq() {
+  const directors = movies.map(function (film) {
+    return film.director
+    }).reduce(function(a,b){
+      if (a.indexOf(b) < 0 ) a.push(b)
+      return a
+    },[])
+  return directors
+  }
+console.log(getAllDirectorsUniq(movies).length) // => 149
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies() {
+  const dramas = movies.filter(function (film) {
+    return film.genre.indexOf('Drama') !== -1
+      })
+  const dramasOfSpielberg = dramas.filter(function (film) {
+    return film.director === 'Steven Spielberg'
+  })
+  return dramasOfSpielberg.length
+}
+console.log(howManyMovies(movies)) // => 4
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage() {
+  const sumOfScores = movies.reduce(function (acc, film) { 
+    return acc + film.score
+  }, 0)
+  return (sumOfScores / movies.length).toFixed(2)
+}
+console.log(scoresAverage(movies)) // => 8.31
```

Comment:
A few things is missing in this function, first you need to pass the argument as mentioned above. You also need to add a check if the array is empty and return 0 in this case. This you could do by adding a check for that `if (movies.length === 0) return 0;` in the first line in the function.

You also need to check when accumulating the score whether this score is defined.  Check the posted solution by Jan.

Finally, toFixed changes the number to string but the test cases expect a number so we should convert the result back to a number `return Number((sumOfScores / movies.length).toFixed(2));`